### PR TITLE
HDDS-7051. Fix offset Condition in ECKeyOutputStream

### PR DESCRIPTION
**Pull request by swamirishi**
## What changes were proposed in this pull request?
Fix offset Condition in ECKeyOutputStream
Condition checking with offset+writtenLen !< length if offset is not zero.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7051

## How was this patch tested?

Current flows all have offset =0.

### CODE REVIEW

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ public void write(byte[] b, int off, int len) throws IOException {
     }
     try {
       int writtenLen = 0;
-      while (off + writtenLen < len) {
```
Comment (umamaheswararao) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java:157]: Thanks for fixing it @swamirishi. This needs a test.

Comment (umamaheswararao) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java:157]: @kaijchen do you want to take a look at it?

Fortunately most of the apps seems to be using off starts from 0, so, we have not hit this one :-)

Comment (kaijchen) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java:157]: LGTM. Thanks for the fix @swamirishi.

**Review by kaijchen — APPROVED** [review]

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java [hunk 2]
```diff
@@ -277,18 +282,28 @@ public void test21ChunksInSingleWriteOp() throws IOException {
     testMultipleChunksInSingleWriteOp(21);
   }
 
-  public void testMultipleChunksInSingleWriteOp(int numChunks)
-      throws IOException {
-    byte[] inputData = getInputBytes(numChunks);
+  public void testMultipleChunksInSingleWriteOp(int offset,
```
Comment (umamaheswararao) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java:285]: This also can be private.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java [hunk 3]
```diff
@@ -277,18 +282,28 @@ public void test21ChunksInSingleWriteOp() throws IOException {
...
-      out.write(inputData);
+      out.write(inputData, offset, numChunks * chunkSize);
     }
 
-    validateContent(inputData, bucket, bucket.getKey(keyName));
+    validateContent(offset, numChunks * chunkSize, inputData, bucket,
+            bucket.getKey(keyName));
+  }
+
+  public void testMultipleChunksInSingleWriteOp(int numChunks)
```
Comment (umamaheswararao) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestECKeyOutputStream.java:304]: This can be private right?

**Review by umamaheswararao — APPROVED** [review]
LGTM, pending CI

**Pull request merged**